solutions/fully-configurable/variables.tf

```diff
@@ -0,0 +1,511 @@
+  description = "The list of access tags to add to the Virtual server instance. For more information, see https://cloud.ibm.com/docs/account?topic=account-access-tags-tutorial."
+  default     = []
+}
+
+##############################################################################
+
+
+##############################################################################
+# VPC Variables
+##############################################################################
+
+variable "existing_vpc_crn" {
+  description = "The CRN of an existing VPC. If the user provides only the `existing_vpc_crn` the VSI will be provisioned in the first subnet of the VPC."
+  type        = string
+  nullable    = false
+}
+
+variable "existing_subnet_id" {
+  description = "The ID of an existing subnet."
+  type        = string
+  default     = null
+}
+
+##############################################################################
+
+
+##############################################################################
+# Virtual server instance Variables
+##############################################################################
+
+variable "vsi_name" {
+  description = "The name of the Virtual server instance."
+  type        = string
+  default     = "vsi"
+}
+
+variable "image_id" {
+  description = "Image ID used for Virtual server instance. Run 'ibmcloud is images' to find available images in a region."
+  type        = string
+  nullable    = false
+}
+
+variable "ssh_public_keys" {
+  description = "List of public SSH keys for Virtual server instance creation which does not already exist in the deployment region. Must be an RSA key with a key size of either 2048 bits or 4096 bits (recommended) - See https://cloud.ibm.com/docs/vpc?topic=vpc-ssh-keys. To use an existing keys, select values for the variable `existing_ssh_key_ids` instead."
+  type        = list(string)
+  default     = []
+
+  validation {
+    error_message = "Public SSH Key must be a valid ssh rsa public key."
+    condition     = alltrue([for ssh in var.ssh_public_keys : can(regex("ssh-rsa AAAA[0-9A-Za-z+/]+[=]{0,3} ?([^@]+@[^@]+)?", ssh))])
+  }
+
+  validation {
+    condition     = var.auto_generate_ssh_key ? true : length(var.ssh_public_keys) > 0 || length(var.existing_ssh_key_ids) > 0 ? true : false
+    error_message = "Please provide a value for either `ssh_public_keys` or `existing_ssh_key_ids`, when `auto_generate_ssh_key` is set to false."
+  }
+}
+
+variable "existing_ssh_key_ids" {
+  description = "The IDs of existing SSH keys to use while creating Virtual server instance."
+  type        = list(string)
+  default     = []
+}
+
+variable "auto_generate_ssh_key" {
+  description = "An SSH key pair (a public and private key) is automatically generated for you. The private key is outputted as an sensitive value which can be stored in the secret mananger. The public key is stored in your VPC and you can download it from the SSH key details page."
+  type        = bool
+  default     = true
```

Comment:
Just checking that this was intentional, having auto_generate default as `true`? I guess I'm not sure what the most common use case will be, I think it depends on the user. If sales/marketing are doing a demo they will choose true here, but an actual customer I'm not sure.

Reply:
I think this is the best user experience, however I think the 3 vars needs to reference each other in their variable descriptions as alternative approaches.